### PR TITLE
api gw 1.16 updates

### PR DESCRIPTION
**Pull request by eddie-rowe**
### Description

Updating API GW docs after testing the latest API GW setup procedures in Kubernetes.

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: website/content/docs/api-gateway/usage/usage.mdx
Comment (eddie-rowe) [reply website/content/docs/api-gateway/usage/usage.mdx:33]: definitely was not intentional - good catch!

File: website/content/docs/api-gateway/configuration/index.mdx
Comment (trujillo-adam) [reply website/content/docs/api-gateway/configuration/index.mdx:22]: @boruszak - "Define a GatewayClass and then configure the parametersRef.name to reference..." is correct, but it sounds strange in my ear. Probably the most technically accurate would be something like:

1. Create a GatewayClass configuration.
2. In the GatewayClass configuration, configure the `parametersRef.name` field to reference the name of your GatewayClassConfig file.

But I feel like we're splitting hairs. I'm fine with your suggestion.

File: website/content/docs/api-gateway/configuration/intentions.mdx
Comment (eddie-rowe) [reply website/content/docs/api-gateway/configuration/intentions.mdx:6]: I like it - I'll update that CodeBlockConfig reference to intentions now. 👍

File: website/content/docs/api-gateway/configuration/index.mdx
Comment (trujillo-adam) [inline website/content/docs/api-gateway/configuration/index.mdx:16]:
```suggestion
- [Intentions](/consul/docs/connect/config-entries/service-intentions) specifies traffic communication rules between services in the mesh. You can also use them specify rules for communication between services outside the mesh that send traffic through Consul API gateway to services in the service mesh.
```
I made another comment directly on the proposed SI conf page, but just be clear, we only want one source of truth for configuration reference. In this case, it should be the existing SI conf entry ref.

File: website/content/docs/api-gateway/configuration/index.mdx
Comment (eddie-rowe) [reply website/content/docs/api-gateway/configuration/index.mdx:22]: Oops - my IDE keeps auto-updating those numbers from repeated 1's to 1,2,3.. 

I haven't created a custom Gateway before, but from what I understand, defining all of these details and applying them together at once is necessary.

**Review by trujillo-adam — CHANGES_REQUESTED** [review]
Thanks for updating this. The main feedback is that we should leverage the existing service intentions configuration entry ref instead of creating a new page. I'm on the same page with Jeff's feedback.

File: website/content/docs/api-gateway/configuration/index.mdx
Comment (eddie-rowe) [reply website/content/docs/api-gateway/configuration/index.mdx:16]: That's fair - I was torn between putting the existing link there and creating a new one. 

I'd like to include the intentions example somewhere, maybe in the upgrade docs, since this is a new workflow with 1.16. What do you think @trujillo-adam ?

Comment (trujillo-adam) [reply website/content/docs/api-gateway/configuration/index.mdx:16]: The convention we started to implement is to add examples for specific use cases to the configuration reference page. https://developer.hashicorp.com/consul/docs/connect/config-entries/service-intentions#examples. The goal is to keep them all in one place as, well, a reference, and link to them as necessary. 
I wouldn't add the example to the upgrade docs, but it is something different that a user has consider when upgrading to 1.16. To that end we could mention that you need to configure intentions in those instructions and link to the example on the reference page.

File: website/content/docs/api-gateway/configuration/intentions.mdx
Comment (eddie-rowe) [reply website/content/docs/api-gateway/configuration/intentions.mdx:1]: I definitely agree there - I'll remove the newly created intentions page from this collection.

File: website/content/docs/api-gateway/configuration/index.mdx
Comment (trujillo-adam) [inline website/content/docs/api-gateway/configuration/index.mdx:16]:
```suggestion
- [Intentions](/consul/docs/connect/config-entries/service-intentions) specify traffic communication rules between services in the mesh. Intentions also enforce rules for service-to-service traffic routed through a Consul API gateway.
```
Sorry, I think this is better than my last suggestion

**Review by trujillo-adam — APPROVED** [review]
Looks good - made one additional suggestion to improve one of my previous suggestions. Approving so you're not blocked.

**Comment by hc-github-team-consul-core** [comment]
After merging, confirm that you see linked PRs AND check them for CI errors.

**Pull request merged**
